feat(guide): port remaining prs/issues from legacy (#10974)
discordjs/discord.js: apps/guide/content/docs/legacy/additional-info/proxy.mdx

---
title: Using a proxy
---

> It may be necessary to use a proxy when using Discord.js, for example when you are deploying your bot on a server that has a web filtering firewall that only allows outside traffic through the proxy. This guide will show you how to set up a proxy with Discord.js.

Reviewer: ```suggestion
It may be necessary to use a proxy when using discord.js, for example when you are deploying your bot on a server that has a web filtering firewall that only allows outside traffic through the proxy. This guide will show you how to set up a proxy with discord.js.
```

title: Using a proxy
---

This guide will show you how to set up a proxy with discord.js. This may be necessary if you are deploying your bot to a server with a firewall only allowing outside traffic through the proxy.

Proxying discord.js requires two components: a REST proxy and a WebSocket proxy.

## Prerequisites

> To achieve these two components you will be utilizing the `undici` and `global-agent` packages:

Reviewer 2: ```suggestion
To achieve these two components you can utilize the `undici` and `global-agent` packages:
```
